core/common/src/LocalTime.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019-2022 JetBrains s.r.o.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime
+
+import kotlinx.datetime.serializers.LocalTimeIso8601Serializer
+import kotlinx.serialization.Serializable
+
+/**
+ * The representation of a specific civil time without a reference to a particular time zone.
+ *
+ * This class does not describe specific *moments in time*, which are represented as [Instant] values.
+ * Instead, its instances can be thought of as clock readings, something that an observer in a particular time zone
+ * could witness.
+ * For example, `18:43` is not a *moment in time*, since someone in Berlin and someone in Tokyo would witness
+ * this on their clocks at different times.
+ *
+ * The main purpose of this class is to provide human-readable representations of [Instant] values, or to transfer them
+ * as data.
+ *
+ * The arithmetic on [LocalTime] values is not provided, since without accounting for the time zone transitions it may give misleading results.
+ */
+@Serializable(LocalTimeIso8601Serializer::class)
+public expect class LocalTime : Comparable<LocalTime> {
+    public companion object {
+
+        /**
+         * Parses a string that represents a time value in ISO-8601 format including time components
+         * but without any time zone component and returns the parsed [LocalTime] value.
+         *
+         * Examples of time in ISO-8601 format:
+         * - `18:43`
+         * - `18:43:00`
+         * - `18:43:00.500`
+         * - `18:43:00.123456789`
+         *
+         * @throws IllegalArgumentException if the text cannot be parsed or the boundaries of [LocalTime] are
+         * exceeded.
+         */
+        public fun parse(isoString: String): LocalTime
+
+        internal val MIN: LocalTime
+        internal val MAX: LocalTime
+    }
+
+    /**
+     * Constructs a [LocalTime] instance from the given time components.
+     *
+     * The supported ranges of components:
+     * - [hour] `0..23`
+     * - [minute] `0..59`
+     * - [second] `0..59`
+     * - [nanosecond] `0..999_999_999`
+     *
+     * @throws IllegalArgumentException if any parameter is out of range.
+     */
+    public constructor(hour: Int, minute: Int, second: Int = 0, nanosecond: Int = 0)
+
+    /** Returns the hour-of-day time component of this date/time value. */
+    public val hour: Int
+    /** Returns the minute-of-hour time component of this date/time value. */
+    public val minute: Int
+    /** Returns the second-of-minute time component of this date/time value. */
+    public val second: Int
+    /** Returns the nanosecond-of-second time component of this date/time value. */
```

Comment:
```suggestion
    /** Returns the nanosecond-of-second time component of this time value. */
```
Ditto for everything else.